### PR TITLE
fix: account for multi-line comments in tokens, improve comment style

### PR DESCRIPTION
**Pull request by jorenbroekema**
*Issue #, if available:*

fixes https://github.com/amzn/style-dictionary/issues/952

*Description of changes:*

- Account for multi-line comments, by splitting them up in the final comment string.
- Put multi-line comments above the token, making it more readable.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jorenbroekema** [comment]
I added a second commit with a small change that allows formatting prop to be user-passed, even if the format is css/sass/less/stylus, as well as a small change to allow passing commentStyle "short-above" or "long-above" to ensure the comment is put 1 line above the outputted property, rather then inline next to it.

Let me know if those two things are okay or you prefer it some other way or via a separate PR

File: __tests__/common/formatHelpers/createPropertyFormatter.test.js [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+      // red = single-line comment, blue = multi-line comment
+      const cssRed = cssFormatter(dictionary.tokens.color.red);
+      const cssBlue = cssFormatter(dictionary.tokens.color.blue);
+      const sassRed = sassFormatter(dictionary.tokens.color.red);
+      const sassBlue = sassFormatter(dictionary.tokens.color.blue);
+
+      // Note that since CSS puts it inside a selector, there is an indentation of 2 spaces as well
+      // CSS also has commentStyle long, whereas sass uses short
+      expect(cssRed).toEqual('  --color-red: #FF0000; /* Foo bar qux */');
```
Comment (dbanksdesign) [inline __tests__/common/formatHelpers/createPropertyFormatter.test.js:78]: Could we make these `toMatchInlineShanpshot` just so if things do change intentionally it is easy to update.

File: lib/common/formatHelpers/createPropertyFormatter.js [hunk 2]
```diff
@@ -19,6 +19,59 @@ const defaultFormatting = {
   suffix: ';'
 }
 
+/**
+ * Split a string comment by newlines and
+ * convert to multi-line comment if necessary
+ * @param {string} to_ret_prop
+ * @param {{comment: string; style: 'short' | 'long'; indentation: string}} options
+ * @param {'short' | 'long'} commentStyle
```
Comment (dbanksdesign) [inline lib/common/formatHelpers/createPropertyFormatter.js:27]: I think this param is unnecessary I think?

File: lib/common/formatHelpers/createPropertyFormatter.js [hunk 3]
```diff
@@ -19,6 +19,59 @@ const defaultFormatting = {
...
+ * @param {string} indentation
+ * @returns {{hasNewLines: boolean, comment: string}}
+ */
+function addComment(to_ret_prop, options) {
+  const { comment, style, indentation } = options;
+
+  const [commentStyle,placeAbove] = style.split('-above');
```
Comment (dbanksdesign) [inline lib/common/formatHelpers/createPropertyFormatter.js:34]: Should we just have a separate option for `commentPosition`? Feels a bit weird to split a string for configuration

File: __tests__/common/formatHelpers/createPropertyFormatter.test.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        "color",
+        "green",
+      ]
+    }
+  }
+};
+const dictionary = createDictionary({ properties });
+
+describe('common', () => {
+  describe('createPropertyFormatter', () => {
```
Comment (dbanksdesign) [inline __tests__/common/formatHelpers/createPropertyFormatter.test.js:63]: Thank you for adding unit tests!

**Review by dbanksdesign — COMMENTED** [review]
Overall looks good, just a few minor things! Thank you!

File: lib/common/formatHelpers/createPropertyFormatter.js (hunk 2 above)
Comment (jorenbroekema) [reply lib/common/formatHelpers/createPropertyFormatter.js]: Indentation param also incorrect, and return type was incorrect too, fixed them

File: lib/common/formatHelpers/createPropertyFormatter.js (hunk 3 above)
Comment (jorenbroekema) [reply lib/common/formatHelpers/createPropertyFormatter.js]: Agree, was a bit hacky, I improved the code to use commentPosition, also improved the switch statement

File: __tests__/common/formatHelpers/createPropertyFormatter.test.js (hunk 1 above)
Comment (jorenbroekema) [reply __tests__/common/formatHelpers/createPropertyFormatter.test.js]: Done!

**Review by dbanksdesign — APPROVED** [review]

**Comment by chris-dura** [comment]
> I added a second commit with a small change that allows formatting prop to be user-passed, even if the format is css/sass/less/stylus, as well as a small change to allow passing commentStyle "short-above" or "long-above" to ensure the comment is put 1 line above the outputted property, rather then inline next to it.
> 
> Let me know if those two things are okay or you prefer it some other way or via a separate PR

Should this have documentation updated as well around the additional options for `commentStyle`?

https://amzn.github.io/style-dictionary/#/formats?id=fileheader

---

Related, are these types incorrect? The types listed here say the options for `commentStyle` are `"short" | "long" | "none"`...

https://github.com/amzn/style-dictionary/blob/0272b8f946bcaef50452af4dc7393b1f559bfdf3/types/FormatHelpers.d.ts#L21

**Pull request merged**